src/inrange.jl

```diff
@@ -32,15 +29,26 @@ function inrange{T <: AbstractFloat}(tree::NNTree{T},
         end
         idxs[i] = idx_in_ball
     end
-    return do_return_inrange(idxs, points)
+    return idxs
 end
 
-do_return_inrange(idxs, ::AbstractVector) = idxs[1]
-do_return_inrange(idxs, ::AbstractMatrix) = idxs
+function inrange{V, T <: Number}(tree::NNTree{V}, point::AbstractVector{T}, radius::Number, sortres=false)
+    idxs = inrange(tree, Vector{T}[point], radius, sortres)
```

Comment:
Should be `Vector{typeof(point)}[point]`.

Also, this one seems a little wasteful. We could copy the kernel from the above function (lines 21-29) - measured about a 10% speedup.